### PR TITLE
Fix numpy bug on Ubuntu16 and Ubuntu18 which will cause segmentfault

### LINKED ISSUE #14780: Ubuntu 18.04+python3.6.7 下 使用paddle跑模型mnist读取数据时报段错误
-Ubuntu 18.04+python3.6.7 下 使用paddle跑模型mnist读取数据时报段错误
- 版本、环境信息：
    1）PaddlePaddle版本：1.1,1.2
    2）CPU：-
    3）GPU：-
    4）系统环境：Ubuntu1804、Ubuntu1604，CentOS7，Python3（3.5.1,3.6.7,3.7.0）
- 模型信息
    1）模型名称 `models`项目中的`mnist`模型
- 复现信息：
     models/fluid/mnist
     python3.6 model.py --batch_size 128 --pass_num 5 --device CPU
- 问题描述：
     运行至reader阶段报错
  

> *** Aborted at 1544089502 (unix time) try "date -d @1544089502" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGSEGV (@0x50) received by PID 182 (TID 0x7f5a294d0700) from PID 80; stack trace: ***
    @     0x7f5a290aa390 (unknown)
    @     0x7f5a292c273c (unknown)
    @     0x7f5a292cb851 (unknown)
    @     0x7f5a292c6564 (unknown)
    @     0x7f5a292cada9 (unknown)
    @     0x7f5a282bf5ad (unknown)
    @     0x7f5a292c6564 (unknown)
    @     0x7f5a282bf664 __libc_dlopen_mode
    @     0x7f5a28291a85 (unknown)
    @     0x7f5a290a7a99 __pthread_once_slow
    @     0x7f5a28291ba4 backtrace
    @     0x7f5a27dff5f3 check_callers.part.0
    @     0x7f5a27dffb20 try_binary_elide
    @     0x7f5a27de794b array_multiply
    @           0x4e6c82 PyNumber_Multiply
    @           0x57403d _PyEval_EvalFrameDefault
    @           0x4f1b93 (unknown)
    @           0x5732d0 _PyEval_EvalFrameDefault
    @           0x4f1b93 (unknown)
    @           0x5e2e1f (unknown)
    @           0x5732d0 _PyEval_EvalFrameDefault
    @           0x57b7fd (unknown)
    @           0x57a7bc (unknown)
    @           0x5730bf _PyEval_EvalFrameDefault
    @           0x572147 (unknown)
    @           0x571ec3 PyEval_EvalCode
    @           0x5e63a2 (unknown)
    @           0x5e680a PyRun_FileExFlags
    @           0x5e65c7 PyRun_SimpleFileExFlags
    @           0x5ebae3 Py_Main
    @           0x4d1ae9 main
    @     0x7f5a2819c830 __libc_start_main
Segmentation fault

### PR DESCRIPTION
**Pull request by velconia**
this fix https://github.com/PaddlePaddle/Paddle/issues/14780

### CODE REVIEW

File: python/paddle/dataset/image.py [hunk 1]
```diff
@@ -32,11 +32,27 @@
 
 from __future__ import print_function
 
+import six
 import numpy as np
-try:
-    import cv2
-except ImportError:
-    cv2 = None
+# NOTE(minqiyang): this is an ugly fix for the numpy bug reported here
```
Comment (typhoonzero) [inline python/paddle/dataset/image.py]: Change to `FIXME:...` and should remove this when numpy have fixed this issue.

**Review by typhoonzero — DISMISSED** [review]

File: python/paddle/dataset/image.py [hunk 2]
```diff
@@ -32,11 +32,28 @@
...
-try:
-    import cv2
-except ImportError:
-    cv2 = None
+# FIXME(minqiyang): this is an ugly fix for the numpy bug reported here
+# https://github.com/numpy/numpy/issues/12497
+if six.PY3:
+    import subprocess
+    import sys
+    import_cv2_proc = subprocess.Popen(
```
Comment (junjun315) [inline python/paddle/dataset/image.py:42]: This code may have problems executing under Windows。make a note

Comment (junjun315) [reply python/paddle/dataset/image.py:42]: checked,it works under Windows

**Review by junjun315 — APPROVED** [review]
LGTM

**Pull request merged**